### PR TITLE
CI: Run on PETSc developement branch

### PR DESCRIPTION
**Pull request by dschwoerer**
Allows to detect issues with PETSc before the release. Should help with bisecting :-)

It would be nicer to only run if PETSc actually changed - but I don't know how to do that

### CODE REVIEW

**Comment by dschwoerer** [comment]
We should probably disable fail-fast if we have failing tests for a long time, to avoid hiding new errors in the other runs.

**Comment by ZedThree** [comment]
I think we should only run this cron job once a week. That should still give us pretty good resolution for breaking changes without using too many resources.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 name: Tests
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '30 6 * * *'
```
Comment (dschwoerer) [inline .github/workflows/tests.yml:6]: @ZedThree It is only one CI job thus I thought daily would gives us very few PRs to look at.
But we can only run it weekly, at which case there are probably like ~10 PRs to go through.
I would rather use computer time but maintainer time.
```suggestion
    # run sunday moring
    - cron: '30 6 * * 0'
```

**Review by ZedThree — APPROVED** [review]

**Comment by ZedThree** [comment]
We might even consider limiting the set of tests to those that actually use PETSc

**Pull request merged**